dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java

```diff
@@ -332,14 +330,14 @@ public Map<String, Object> queryDataSourceList(User loginUser, Integer type) {
             Set<Integer> ids = resourcePermissionCheckService
                     .userOwnedResourceIdsAcquisition(AuthorizationType.DATASOURCE, loginUser.getId(), log);
             if (ids.isEmpty()) {
-                result.put(Constants.DATA_LIST, Collections.emptyList());
+                result.setData(Collections.emptyList());
                 putMsg(result, Status.SUCCESS);
                 return result;
             }
             datasourceList = dataSourceMapper.selectBatchIds(ids).stream()
                     .filter(dataSource -> dataSource.getType().getCode() == type).collect(Collectors.toList());
         }
-        result.put(Constants.DATA_LIST, datasourceList);
+        result.setData(datasourceList);
```

Comment:
Maybe I misrepresented myself earlier, I want to said is we should return the basic type in service and then wrapper it in the controller. Because someone will re-run the result of service so we should keep it as simple as possible. In this case, we should return `datasourceList` in method `queryDataSourceList` and then we wrapper `queryDataSourceList` into `Result` in controller by using `Result.success(datasourceList)`